core/src/main/java/org/infinispan/eviction/impl/PassivationManagerImpl.java

```diff
@@ -99,14 +101,14 @@ public void passivateAll() throws PersistenceException {
       if (enabled && !skipOnStop) {
          long start = timeService.time();
          log.passivatingAllEntries();
-         long count = 0;
-         for (InternalCacheEntry e : container) {
-            count++;
-            if (trace) log.tracef("Passivating %s", e.getKey());
-            persistenceManager.writeToAllNonTxStores(marshalledEntryFactory.newMarshalledEntry(e.getKey(), e.getValue(),
-                                                                                               internalMetadata(e)), BOTH);
-         }
-         log.passivatedEntries(count, Util.prettyPrintTime(timeService.timeDuration(start, TimeUnit.MILLISECONDS)));
+
+         final AtomicInteger count = new AtomicInteger();
```

Comment:
This unfortunately won't work correctly if we have more than 1 cache writer. I would just store `container.sizeIncludingExpired` and pass that to the log message below. It can be off if we have expired entries, but it should be fine really.